### PR TITLE
test: add full cairo 1 test flow

### PR DESCRIPTION
**Pull request by LucasLvy**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

-
-
-

### CODE REVIEW

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

File: tests/tests/test-rpc/test-starknet-rpc.ts [hunk 1]
```diff
@@ -2009,4 +2010,106 @@ describeDevMadara("Starknet RPC", (context) => {
...
+        "latest"
+      );
+      // TODO: (Apoorv) make these checks better once we to_rpc_contract_class is fixed #775 and #790
+      expect(contractClassActual).to.have.property("entry_points_by_type");
+      expect(contractClassActual).to.have.property("sierra_program");
+      expect(contractClassActual).to.have.property("contract_class_version");
+      expect(contractClassActual).to.have.property("abi");
+      expect(res.class_hash).to.be.eq(classHash);
+
+      const deployRes = await account.deploy(
```
Comment (EvolveArt) [inline tests/tests/test-rpc/test-starknet-rpc.ts:2050]: `deployContract` ?

File: tests/tests/test-rpc/test-starknet-rpc.ts [hunk 2]
```diff
@@ -2009,4 +2010,106 @@ describeDevMadara("Starknet RPC", (context) => {
...
+
+      const balanceSender = await providerRPC.getStorageAt(
+        deployRes.contract_address[0],
+        "0x617243ac31335377b9d26d1a6b02f47b419ad593e1ae67660dd27ec77635906",
+        "latest"
+      );
+      expect(toHex(balanceSender)).to.be.equal("0x0");
+      const balanceRecipient = await providerRPC.getStorageAt(
+        deployRes.contract_address[0],
+        "0x753d37842b9cfa00ee311ab2564951681d89ee4d5596e84e74030de35018c8a",
```
Comment (EvolveArt) [inline tests/tests/test-rpc/test-starknet-rpc.ts:2107]: add how you computed this in comment

File: tests/tests/test-rpc/test-starknet-rpc.ts [hunk 3]
```diff
@@ -2009,4 +2010,106 @@ describeDevMadara("Starknet RPC", (context) => {
       expect(latestBlock.new_root).to.eq("0x0");
     });
   });
+
+  describe("Cairo 1 full flow", async () => {
+    it("should declare erc20 contract", async () => {
```
Comment (EvolveArt) [inline tests/tests/test-rpc/test-starknet-rpc.ts:2015]: description `declare, deploy and test storage of an erc20 contract`

**Review by EvolveArt — COMMENTED** [review]
Maybe we should start splitting the integration tests into multiple files, wdyt ?

**Comment by LucasLvy** [comment]
Good idea can you create the issue ?

File: tests/tests/test-rpc/test-starknet-rpc.ts (hunk 1 above)
Comment (LucasLvy) [reply tests/tests/test-rpc/test-starknet-rpc.ts:2050]: no

**Pull request merged**